src/sentry/migrations/0150_remove_slack_workspace_orgintegrations.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.29 on 2021-01-20 19:29
+from __future__ import unicode_literals
+
+from django.db import migrations
+from django.db.models import Q
+
+
+def remove_slack_workspace_apps(apps, schema_editor):
+    """
+    Remove OrganizationIntegrations that are linked to Slack Integrations that are
+    still using workspace apps.
+
+    We can determine which integrations are workspace apps by looking at the metadata.
+    There is a field for new or migrated slack integrations called the `installation_type`.
+    If a slack integration is still a workspace app, it will not have this field
+    in the metadata.
+    """
+
+    Integration = apps.get_model("sentry", "Integration")
+    OrganizationIntegration = apps.get_model("sentry", "OrganizationIntegration")
+
+    integrations = (
+        Integration.objects.all()
+        .filter(provider="slack")
+        .exclude(metadata__icontains="installation_type")
+        .values_list("id", flat=True)
+    )
+
+    for org_integration in OrganizationIntegration.objects.filter(integration_id__in=integrations):
```

Comment:
We should use the `RangeQuerySetWrapperWithProgressBar` for this.

Reply:
I thought about that, I didn't think it was a terribly large table but not sure "too many rows" is for using filter. Can update though

Reply:
Never can be too safe. About how many rows are there?

Reply:
less than 100k

Reply:
👍 for using RangeQuerySetWrapperWithProgressBar here.

Reply:
I'm also not sure how many rows are "too many" before recommending `BulkDeleteQuery`. It seems to default to delete in batches of 100k, so probably not worth it here.